src/test/java/com/google/neighborgood/IntegrationTest.java

```diff
@@ -0,0 +1,1065 @@
+    verifyNewTaskHomepage();
+  }
+
+  @Test
+  /** Tests functionality of adding tasks from the userpage */
+  public void _03_UserPage_AsLoggedUser_AddTask() {
+    goToUserPage();
+    ifLaggingThenRefresh();
+
+    // Randomizes task contents
+    Random random = new Random();
+    String taskDetail = TASK_DETAIL + random.nextInt(1000);
+    String taskOverview = TASK_OVERVIEW + random.nextInt(1000);
+    String rewardPoints = Integer.toString(random.nextInt(201));
+    int categoryOptionIndex = random.nextInt(TASK_CATEGORIES.length);
+    String taskCategory = TASK_CATEGORIES[categoryOptionIndex];
+
+    addTask(taskDetail, rewardPoints, categoryOptionIndex, taskOverview);
+
+    // User should be redirected to user profile page after adding a task
+    assertTrue(
+        "User should be in user profile page",
+        driver.getCurrentUrl().contains("/user_profile.jsp"));
+
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+    verifyNewTaskUserPage();
+    backToHome();
+    ifLaggingThenRefresh();
+    verifyNewTaskHomepage();
+  }
+
+  @Test
+  /** Tests functionality of logging out */
+  public void _04_Homepage_AsLoggedUser_LogOut() {
+    // Logs out by using 'loginLogoutMessage' element id;
+    logOut("loginLogoutMessage");
+    String loginMessageElement =
+        wait.until(
+                new Function<WebDriver, WebElement>() {
+                  public WebElement apply(WebDriver driver) {
+                    return driver.findElement(By.id("loginLogoutMessage"));
+                  }
+                })
+            .getText();
+    assertEquals("Login to help out a neighbor!", loginMessageElement);
+  }
+
+  @Test
+  /** Tests functionality of a helper helping out with a task */
+  public void _05_Homepage_AsLoggedHelper_HelpOut() {
+    loginNewUser(
+        USER_EMAIL_HELPER,
+        USER_NICKNAME_HELPER,
+        USER_ADDRESS,
+        USER_PHONE,
+        USER_ZIPCODE,
+        USER_COUNTRY);
+    ifLaggingThenRefresh();
+    // Verifies logged user in userpage and then in homepage
+    verifyLoggedUser(USER_NICKNAME_HELPER, "log-out-link");
+    backToHome();
+    ifLaggingThenRefresh();
+    verifyLoggedUser(USER_NICKNAME_HELPER, "login-logout");
+    helpOut();
+    goToUserPage();
+    ifLaggingThenRefresh();
+    goToOfferHelp();
+    verifyOfferHelpTask();
+  }
+
+  @Test
+  /** Tests functionality of having a helper mark a task as complete */
+  public void _06_Userpage_AsLoggedHelper_CompleteTask() {
+    completeTaskAsHelper();
+    driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+
+    // Refreshes page due to flakiness of test caused by the partial refresh of the div in the page.
+    // Without refreshing this often resulted in a a stale element error
+    driver.navigate().refresh();
+    goToOfferHelp();
+
+    // Location of first task listed in the complete task table in userpage
+    String taskCompletedXPath = "//tbody[@id='complete-task-body']/tr[1]";
+
```

Comment:
Can we use a for loop here instead of having so many duplicated code? For example, how about creating an array of String such as `String[] testTargets = new String[]{"overview", "status", "nickname" ... etc.};` Then create a for loop, during the loop we set a value of a String variable `temp` to the result of `wait.until` and make an assertion on the variable `temp` during each loop. The change of `/td[i]` can also be achieved by using the index.

Using a loop can save a lot of spaces and make the code cleaner. 😄

Reply:
This is a great suggestion, thanks! I'll make these changes now